app/connectors/UploadConnector.scala

```diff
@@ -16,54 +16,122 @@
+                                val configuration: Configuration,
+                                environment: Environment)
+                               (implicit ec: ExecutionContext)
+  extends ServicesConfig {
 
   override protected def mode: Mode = environment.mode
 
   override protected def runModeConfiguration: Configuration = configuration
 
   implicit val hc: HeaderCarrier = HeaderCarrier()
-  val serviceUrl = baseUrl("voa-bar")
-  val baseSegment = "/voa-bar/"
-  val xmlContentTypeHeader = ("Content-Type", "text/plain")
+  private[connectors] val serviceUrl = baseUrl("voa-bar")
+  private[connectors] val baseSegment = "/voa-bar/"
+  private[connectors] val xmlContentTypeHeader = ("Content-Type", "text/plain")
+  private[connectors] val upScanConfig = configuration.getConfig("microservice.services.upscan").get
+  private[connectors] val upScanPort = upScanConfig.getString("port").get
+  private[connectors] val upScanHost = upScanConfig.getString("host").get
+  private[connectors] val initiateUrl = s"http://$upScanHost:$upScanPort${upScanConfig.getString("initiate.url").get}"
 
   def generateUsernameHeader(username: String) = ("BA-Code", username)
 
   def generatePasswordHeader(password: String) = ("password", password)
 
-  def sendXml(xml: String, loginDetails: Login): Future[Try[String]] = {
+  def sendXml(xml: String, loginDetails: Login): Future[Either[Error, String]] = {
     val baCode = loginDetails.username
     val password = loginDetails.password
     http.POST(s"$serviceUrl${baseSegment}upload", xml, Seq(xmlContentTypeHeader, generateUsernameHeader(baCode), generatePasswordHeader(password)))
       .map {
         response =>
           response.status match {
-            case 200 => Success(response.body)
+            case Status.OK => Right(response.body)
             case status => {
-              Logger.warn("Received status of " + status + " from upstream service when uploading am xml file")
-              Failure(new RuntimeException("Received status of " + status + " from upstream service when uploading an xml file"))
+              val errorMsg = s"Received status of $status from upstream service when uploading am xml file"
+              Logger.warn(errorMsg)
+              Left(Error("councilTaxUpload.error.fileUploadService", Seq(errorMsg)))
             }
           }
       } recover {
       case e =>
-        Logger.warn("Received exception " + e.getMessage + " from upstream service when uploading am xml file")
-        Failure(new RuntimeException("Received exception " + e.getMessage + " from upstream service when uploading am xml file"))
+        val errorMsg = s"Received status of ${e.getMessage} from upstream service when uploading am xml file"
+        Logger.error(errorMsg, e)
+        Left(Error("councilTaxUpload.error.fileUploadService", Seq(errorMsg)))
     }
   }
 
+  def initiate(request: InitiateRequest): Future[Either[Error, InitiateResponse]] = {
```

Comment:
all these three are upscan methods to get the credentials, upload a file and download a file